src/quickgui/qgsquickmapsettings.cpp

```diff
@@ -0,0 +1,214 @@
+ *                                                                         *
+ ***************************************************************************/
+
+
+#include "qgsmaplayer.h"
+#include "qgsmaplayerstylemanager.h"
+#include "qgsmessagelog.h"
+#include "qgsproject.h"
+#include "qgis.h"
+
+#include "qgsquickmapsettings.h"
+
+QgsQuickMapSettings::QgsQuickMapSettings( QObject *parent )
+  : QObject( parent )
+{
+  // Connect signals for derived values
+  connect( this, &QgsQuickMapSettings::destinationCrsChanged, this, &QgsQuickMapSettings::mapUnitsPerPixelChanged );
+  connect( this, &QgsQuickMapSettings::extentChanged, this, &QgsQuickMapSettings::mapUnitsPerPixelChanged );
+  connect( this, &QgsQuickMapSettings::outputSizeChanged, this, &QgsQuickMapSettings::mapUnitsPerPixelChanged );
+  connect( this, &QgsQuickMapSettings::extentChanged, this, &QgsQuickMapSettings::visibleExtentChanged );
+  connect( this, &QgsQuickMapSettings::rotationChanged, this, &QgsQuickMapSettings::visibleExtentChanged );
+  connect( this, &QgsQuickMapSettings::outputSizeChanged, this, &QgsQuickMapSettings::visibleExtentChanged );
+}
+
+void QgsQuickMapSettings::setProject( QgsProject *project )
+{
+  if ( project == mProject )
+    return;
+
+  // If we have already something connected, disconnect it!
+  if ( mProject )
+  {
+    mProject->disconnect( this );
+  }
+
+  mProject = project;
+
+  // Connect all signals
+  if ( mProject )
+  {
+    connect( mProject, &QgsProject::readProject, this, &QgsQuickMapSettings::onReadProject );
+    setDestinationCrs( mProject->crs() );
+    mMapSettings.setTransformContext( mProject->transformContext() );
+  }
+  else
+  {
+    mMapSettings.setTransformContext( QgsCoordinateTransformContext() );
+  }
+
+  emit projectChanged();
+}
+
+QgsProject *QgsQuickMapSettings::project() const
+{
+  return mProject;
+}
+
+QgsCoordinateTransformContext QgsQuickMapSettings::transformContext() const
+{
+  return mMapSettings.transformContext();
+}
+
+QgsRectangle QgsQuickMapSettings::extent() const
+{
+  return mMapSettings.extent();
+}
+
+void QgsQuickMapSettings::setExtent( const QgsRectangle &extent )
+{
+  if ( mMapSettings.extent() == extent )
+    return;
+
+  mMapSettings.setExtent( extent );
+  emit extentChanged();
+}
+
+void QgsQuickMapSettings::setCenter( const QgsPoint &center )
+{
+  QgsVector delta = QgsPointXY( center ) - mMapSettings.extent().center();
+
+  QgsRectangle e = mMapSettings.extent();
+  e.setXMinimum( e.xMinimum() + delta.x() );
+  e.setXMaximum( e.xMaximum() + delta.x() );
+  e.setYMinimum( e.yMinimum() + delta.y() );
+  e.setYMaximum( e.yMaximum() + delta.y() );
+
+  setExtent( e );
+}
+
+double QgsQuickMapSettings::mapUnitsPerPixel() const
+{
+  return mMapSettings.mapUnitsPerPixel();
+}
+
+QgsRectangle QgsQuickMapSettings::visibleExtent() const
+{
+  return mMapSettings.visibleExtent();
+}
+
+QPointF QgsQuickMapSettings::coordinateToScreen( const QgsPoint &p ) const
```

Comment:
Can you make those single-character variable names longer?

Reply:
done